sunpy/visualization/animator/image.py

```diff
@@ -220,28 +221,63 @@ def __init__(self, data, wcs=None, image_axes=[-1, -2], unit_x_axis=None, unit_y
     def _get_main_axes(self):
         axes = self.fig.add_axes([0.1, 0.1, 0.8, 0.8], projection=self.wcs,
                                  slices=self.slices_wcsaxes)
-        self._set_unit_in_axis(axes)
         return axes
 
-    def _set_unit_in_axis(self, axes):
-        if self.unit_x_axis is not None:
-            axes.coords[2].set_format_unit(self.unit_x_axis)
-            axes.coords[2].set_ticks(exclude_overlapping=True)
-        if self.unit_y_axis is not None:
-            axes.coords[1].set_format_unit(self.unit_y_axis)
-            axes.coords[1].set_ticks(exclude_overlapping=True)
-
     def plot_start_image(self, ax):
         """Sets up plot of initial image."""
         imshow_args = {'interpolation': 'nearest',
                        'origin': 'lower',
                        }
         imshow_args.update(self.imshow_kwargs)
+        ax.ticklabel_format(style='sci')
         im = ax.imshow(self.data[self.frame_index], **imshow_args)
         if self.if_colorbar:
             self._add_colorbar(im)
         return im
 
+    def update_axes_style(self, tick_spacing, x_tick_position=None,
+                          y_tick_position=None,  xlabel=None, ylabel=None):
+        """
+        Allows a user to change the axes' units, labels and ticks of an existing visualization.
+
+        Parameters
+        ----------
+
+        tick_spacing: `astropy.units`
+            Spacing between two consecutive ticks
+        x_tick_position: `str`
+            Tick position for the X axis
+        y_tick_position: `str`
+            Tick position for the Y axis
+        xlabel: `str`
+            Label for X axis in the plot
+        ylabel: `str`
+            Label for Y axis in the plot
+
+        """
+        ax = self.axes
+        for i, axis_units in enumerate([self.unit_y_axis, self.unit_x_axis], start=1):
+            if axis_units is not None:
+                ax.coords[i].set_format_unit(axis_units)
+
+        if not isinstance(tick_spacing, u.Quantity):
+            raise TypeError("Astropy Quantities should be provided")
+        else:
+            for i in [1, 2]:
```

Comment:
This shouldn't be hard coded to 1 and 2, it could be any two axes of the cube depending on `image_axes`.